boto/auth.py

```diff
@@ -1038,13 +1062,33 @@ def _wrapper(self):
         if boto.config.get('s3', 'use-sigv4', False):
             return ['hmac-v4-s3']
 
-        if hasattr(self, 'host'):
-            # If you're making changes here, you should also check
-            # ``boto/iam/connection.py``, as several things there are also
-            # endpoint-related.
-            for test in SIGV4_DETECT:
-                if test in self.host:
-                    return ['hmac-v4-s3']
-
-        return func(self)
+        if not hasattr(self, 'host'):
+            return func(self)
+
+        # Use default for non-aws hosts. Adding a url scheme is necessary if
+        # not present for urlparse to properly function.
+        host = self.host
+        if not self.host.startswith('http://') or \
+                self.host.startswith('https://'):
+            host = 'https://' + host
+        netloc = urlparse(host).netloc
+        if not (netloc.endswith('amazonaws.com') or
+                netloc.endswith('amazonaws.com.cn')):
```

Comment:
This is a bit more robust than before in that a customer with a domain like `amazonaws.com.mydomain.com` or somesuch won't hit the sigv4 switch, but now we have to account for every tld that amazon is using. So do we foresee us having more tld's besides those? We haven't done so for any other country.